src/sinks/aws_kinesis/firehose/config.rs

```diff
@@ -180,4 +186,13 @@ impl RetryLogic for KinesisRetryLogic {
         }
         is_retriable_error(error)
     }
+
+    fn should_retry_response(&self, response: &Self::Response) -> RetryAction {
+        if self.retry_partial && response.failure_count > 0 {
```

Comment:
nit: you could flip the logic on this conditional to evaluate if there were failures first, since when retry_partial is set to true it would always be true even on successful responses.

(same applies for the other implementation of `should_retry_response()`

Reply:
Fixed